### PR TITLE
WFLY-18352 Optimize metadata mapping for distributed timers

### PR DESCRIPTION
**Pull request by pferraro**
https://issues.redhat.com/browse/WFLY-18352

Requires https://github.com/wildfly/wildfly/pull/17096

### CODE REVIEW

**Comment by pferraro** [comment]
Rebased against main.

File: clustering/ejb/cache/src/main/java/org/wildfly/clustering/ejb/cache/timer/TimerMetaDataEntryFunction.java [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+import java.time.Duration;
+import java.util.function.Supplier;
+
+import org.wildfly.clustering.ee.cache.function.RemappingFunction;
+import org.wildfly.clustering.ee.cache.offset.Offset;
+import org.wildfly.clustering.ee.cache.offset.OffsetValue;
+import org.wildfly.common.function.Functions;
+
+/**
```
Comment (rachmatowicz) [inline clustering/ejb/cache/src/main/java/org/wildfly/clustering/ejb/cache/timer/TimerMetaDataEntryFunction.java:16]: Purpose?

File: clustering/ejb/infinispan/src/main/java/org/wildfly/clustering/ejb/infinispan/timer/InfinispanTimerManagerConfiguration.java [hunk 2]
```diff
@@ -18,11 +20,11 @@
 /**
```
Comment (rachmatowicz) [inline clustering/ejb/infinispan/src/main/java/org/wildfly/clustering/ejb/infinispan/timer/InfinispanTimerManagerConfiguration.java:20]: Purpose?

File: clustering/ejb/infinispan/src/main/java/org/wildfly/clustering/ejb/infinispan/timer/InfinispanTimerMetaDataFactory.java [hunk 3]
```diff
@@ -6,70 +6,77 @@
...
+import org.wildfly.clustering.ejb.cache.timer.TimerIndex;
+import org.wildfly.clustering.ejb.cache.timer.TimerIndexKey;
+import org.wildfly.clustering.ejb.cache.timer.TimerMetaDataConfiguration;
+import org.wildfly.clustering.ejb.cache.timer.TimerMetaDataEntryFunction;
+import org.wildfly.clustering.ejb.cache.timer.TimerMetaDataFactory;
+import org.wildfly.clustering.ejb.cache.timer.TimerMetaDataKey;
 import org.wildfly.clustering.ejb.timer.ImmutableTimerMetaData;
 import org.wildfly.clustering.ejb.timer.TimerMetaData;
 
 /**
```
Comment (rachmatowicz) [inline clustering/ejb/infinispan/src/main/java/org/wildfly/clustering/ejb/infinispan/timer/InfinispanTimerMetaDataFactory.java:30]: Purpose?

File: clustering/ejb/infinispan/src/main/java/org/wildfly/clustering/ejb/infinispan/timer/TimerMetaDataKeyFilter.java [hunk 4]
```diff
@@ -6,15 +6,16 @@
 package org.wildfly.clustering.ejb.infinispan.timer;
 
 import org.infinispan.util.function.SerializablePredicate;
+import org.wildfly.clustering.ejb.cache.timer.TimerMetaDataKey;
 
 /**
```
Comment (rachmatowicz) [inline clustering/ejb/infinispan/src/main/java/org/wildfly/clustering/ejb/infinispan/timer/TimerMetaDataKeyFilter.java:11]: Purpose?

File: clustering/ejb/infinispan/src/main/java/org/wildfly/clustering/ejb/infinispan/timer/TimerScheduler.java [hunk 5]
```diff
@@ -38,20 +40,20 @@
 /**
```
Comment (rachmatowicz) [inline clustering/ejb/infinispan/src/main/java/org/wildfly/clustering/ejb/infinispan/timer/TimerScheduler.java:40]: Purpose?

**Review by rachmatowicz — APPROVED** [review]
Looked over the changes implemented to consolidate metadata in distributed timers. Looks good, with extensive unit testing.

File: clustering/ejb/cache/src/main/java/org/wildfly/clustering/ejb/cache/timer/TimerMetaDataEntryFunction.java (hunk 1 above)
Comment (pferraro) [reply clustering/ejb/cache/src/main/java/org/wildfly/clustering/ejb/cache/timer/TimerMetaDataEntryFunction.java:16]: Will add missing javadoc.

File: clustering/ejb/infinispan/src/main/java/org/wildfly/clustering/ejb/infinispan/timer/InfinispanTimerManagerConfiguration.java (hunk 2 above)
Comment (pferraro) [reply clustering/ejb/infinispan/src/main/java/org/wildfly/clustering/ejb/infinispan/timer/InfinispanTimerManagerConfiguration.java:20]: Will add missing javadoc.

File: clustering/ejb/infinispan/src/main/java/org/wildfly/clustering/ejb/infinispan/timer/InfinispanTimerMetaDataFactory.java (hunk 3 above)
Comment (pferraro) [reply clustering/ejb/infinispan/src/main/java/org/wildfly/clustering/ejb/infinispan/timer/InfinispanTimerMetaDataFactory.java:30]: Will add missing javadoc.

File: clustering/ejb/infinispan/src/main/java/org/wildfly/clustering/ejb/infinispan/timer/TimerMetaDataKeyFilter.java (hunk 4 above)
Comment (pferraro) [reply clustering/ejb/infinispan/src/main/java/org/wildfly/clustering/ejb/infinispan/timer/TimerMetaDataKeyFilter.java:11]: Will add missing javadoc.

File: clustering/ejb/infinispan/src/main/java/org/wildfly/clustering/ejb/infinispan/timer/TimerScheduler.java (hunk 5 above)
Comment (pferraro) [reply clustering/ejb/infinispan/src/main/java/org/wildfly/clustering/ejb/infinispan/timer/TimerScheduler.java:40]: Will add missing javadoc.

**Pull request merged**